common/src/main/java/com/wynntils/core/chat/transcoder/CodedStyle.java

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.chat.transcoder;
+
+import com.wynntils.utils.colors.CustomColor;
+import java.util.Arrays;
+import java.util.Optional;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.HoverEvent;
+import net.minecraft.network.chat.Style;
+
+public final class CodedStyle {
+    private static final String STYLE_PREFIX = "§";
+
+    private final CodedStringPart owner;
+
+    private final CustomColor color;
+    private final boolean bold;
+    private final boolean italic;
+    private final boolean underlined;
+    private final boolean strikethrough;
+    private final boolean obfuscated;
+    private final ClickEvent clickEvent;
+    private final HoverEvent hoverEvent;
+
+    private Style styleCache;
+
+    private CodedStyle(
+            CodedStringPart owner,
+            CustomColor color,
+            boolean bold,
+            boolean italic,
+            boolean underlined,
+            boolean strikethrough,
+            boolean obfuscated,
+            ClickEvent clickEvent,
+            HoverEvent hoverEvent) {
+        this.owner = owner;
+        this.color = color;
+        this.bold = bold;
+        this.italic = italic;
+        this.underlined = underlined;
+        this.strikethrough = strikethrough;
+        this.obfuscated = obfuscated;
+        this.clickEvent = clickEvent;
+        this.hoverEvent = hoverEvent;
+    }
+
+    static CodedStyle fromStyle(Style style, CodedStringPart owner, CodedStringPart partBefore) {
```

Comment:
Actually, I think you don't need a CodedStringPart here. All you need is a `CodedStyle accumulatedStyle`. I think it will be clearer if you limit it this way.